app/code/Magento/Catalog/Model/ResourceModel/Category/Collection.php

```diff
@@ -6,12 +6,14 @@
 namespace Magento\Catalog\Model\ResourceModel\Category;
 
 use Magento\CatalogUrlRewrite\Model\CategoryUrlRewriteGenerator;
+use Magento\Store\Model\ScopeInterface;
 
 /**
  * Category resource collection
  *
  * @api
  * @author      Magento Core Team <core@magentocommerce.com>
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
```

Comment:
Adding suppressions for CouplingBetweenObjects warnings is not allowed. The reason is that this increases Technical Debt. We understand this means more work, but refactoring this will help improve the code. Also, just adding an additional parameter would change the signature of the `@api` annotated class. Both changes mean that we probably should rebase this PR against `2.3-develop` so we can add breaking changes.

Reply:
I will do a PR on 2.3-develop. 

The suppression is the same as the one from it's parent, so I'll keep that one.